sdk/core/core/src/az_span.c

```diff
@@ -618,3 +618,88 @@ _az_span_scan_until(az_span self, _az_predicate predicate, int32_t* out_index)
   }
   return AZ_ERROR_ITEM_NOT_FOUND;
 }
+
+AZ_NODISCARD az_span _az_span_trim_white_space(az_span source)
```

Comment:
I would think that this whole function could just be implemented in 1 line of code:

return _az_span_trim_white_space_from_end(_az_span_trim_white_space_from_start(source));

Reply:
Yep, an alternative makes the other two methods 1 liners:
https://github.com/Azure/azure-sdk-for-c/pull/637#discussion_r417694196